test/foundry/IdleCDOAutoLossVariant.t.sol

```diff
@@ -0,0 +1,364 @@
+    uint256 amount = 10000 * ONE_SCALE;
+    // fee is set to 10% and release block period to 0
+    (uint256 preAAPrice, uint256 preBBPrice) = _doDepositsWithInterest(amount, amount);
+
+    uint256 currPrice = strategy.price();
+    uint256 maxDecrease = IdleCDOAutoLossVariant(address(idleCDO)).maxDecreaseDefault();
+    uint256 unclaimedFees = idleCDO.unclaimedFees();
+    // now let's simulate a loss by decreasing strategy price
+    // curr price - 5%
+    vm.mockCall(
+      address(strategy),
+      abi.encodeWithSelector(IIdleCDOStrategy.price.selector),
+      abi.encode(currPrice * (FULL_ALLOC - maxDecrease) / FULL_ALLOC)
+    );
+
+    uint256 postAAPrice = idleCDO.virtualPrice(address(AAtranche));
+    uint256 postBBPrice = idleCDO.virtualPrice(address(BBtranche));
+    // juniors lost about 10% as there were seniors to cover
+    assertApproxEqAbs(
+      preBBPrice * 90000/100000,
+      postBBPrice, 
+      200 // 200 wei to account for interest accrued
+    );
+    // seniors are covered
+    assertEq(preAAPrice, postAAPrice, 'AA price unaffected');
+    assertEq(idleCDO.priceAA(), preAAPrice, 'AA price not updated until new interaction');
+    assertEq(idleCDO.priceBB(), preBBPrice, 'BB price not updated until new interaction');
+
+    _depositWithUser(idleCDO.rebalancer(), amount, true);
+    _depositWithUser(idleCDO.rebalancer(), amount, false);
+
+    uint256 postDepositAAPrice = idleCDO.virtualPrice(address(AAtranche));
+    uint256 postDepositBBPrice = idleCDO.virtualPrice(address(BBtranche));
+
+    assertEq(postDepositAAPrice, postAAPrice, 'AA price did not change after deposit');
+    assertEq(postDepositBBPrice, postBBPrice, 'BB price did not change after deposit');
+    assertEq(idleCDO.priceAA(), postDepositAAPrice, 'AA saved price updated');
+    assertEq(idleCDO.priceBB(), postDepositBBPrice, 'BB saved price updated');
+
+    assertEq(idleCDO.unclaimedFees(), unclaimedFees, 'Fees did not increase');
+    vm.clearMockedCalls();
+  }
+
+  function testRedeemWithLossCovered() external runOnForkingNetwork(MAINNET_CHIANID) {
+    uint256 maxDecrease = IdleCDOAutoLossVariant(address(idleCDO)).maxDecreaseDefault();
+    uint256 amount = 10000 * ONE_SCALE;
+    // fee is set to 10% and release block period to 0, AARatio 66%
+    (uint256 preAAPrice, uint256 preBBPrice) = _doDepositsWithInterest(amount * 2, amount);
+
+    // now let's simulate a loss by decreasing cToken price (so to decrease strategy.price), curr price - 5%
+    vm.mockCall(
+      address(cToken),
+      abi.encodeWithSelector(ICToken.exchangeRateStored.selector),
+      abi.encode(ICToken(cToken).exchangeRateStored() * (FULL_ALLOC - maxDecrease) / FULL_ALLOC)
+    );
+
+    uint256 postAAPrice = idleCDO.virtualPrice(address(AAtranche));
+    uint256 postBBPrice = idleCDO.virtualPrice(address(BBtranche));
+    // juniors lost about 15% as there were 2x seniors to cover
```

Comment:
i'm not sure how to calculate juniors lose (15%)

Reply:
AAtranche total supply is two times BBtranche supply.
-> juniors lost 5+5*2 = 15 % ?

Reply:
yes